lineage/lineage_graph.py

```diff
@@ -302,6 +303,22 @@ def _enrich_graph_with_monitoring_context(self) -> None:
                     </body>
                 </html>
                 """
+
+                node_volumes = self.catalog[node]['volume']
+                if node_volumes[-1] < median(node_volumes) / 2:
```

Comment:
Nice!
Similarly, possible to use Strategy Pattern here as well. Decoupling the "anomaly detection" implementation from the display itself

I would consider using the [Tukey's fence method](https://en.wikipedia.org/wiki/Outlier#Tukey's_fences), where the value is greater/lesser than 1.5 x the Inter Quartile Range